### PR TITLE
Feature: sorted deps

### PR DESCRIPTION
**Pull request by iamstarkov**
`npm` [installs things][demo] in [sorted manner][sorted], `pnpm` doesn't, so if you use `npm` after `pnpm`, you will get rewrited `package.json` and messy diff. This is a fix

[demo]: https://github.com/iamstarkov/npm-install-sorted-demo
[sorted]: https://github.com/npm/npm/blob/master/lib/utils/deep-sort-object.js#L2

### CODE REVIEW

**Comment by iamstarkov** [comment]
it turned out that CI tests are failing only on `saveDev scoped module to package.json (@rstacruz/tap-spec)` and unfortunately all scoped related are not passing on my machine because of timeout

**Comment by iamstarkov** [comment]
it seems like test cases are not isolated

**Comment by iamstarkov** [comment]
do you have any idea why scoped modules can not work? separate `$ npm i -D @rstacruz/tap-spec` works fine

**Comment by iamstarkov** [comment]
except problems with testing scoped modules, tests are green and feature is working

**Comment by rstacruz** [comment]
> do you have any idea why scoped modules can not work? separate $ npm i -D @rstacruz/tap-spec works fine

it should, actually. did you ever figure it out?

File: test/index.js [hunk 1]
```diff
@@ -209,9 +209,9 @@ test('saveDev scoped module to package.json (@rstacruz/tap-spec)', function (t)
   }, t.end)
 })
 
-test('multiple save to package.json with `exact` versions (@rstacruz/tap-spec & rimraf@2.5.1)', function (t) {
+test.only('multiple save to package.json with `exact` versions (@rstacruz/tap-spec & rimraf@2.5.1) (in sorted order)', function (t) {
```
Comment (LinusU) [inline test/index.js]: `.only` probably shouldn't be part of this pr ;)

Comment (rstacruz) [inline test/index.js]: lmao, i'm going to add `grep -v test.only test/*.js` to this repo's travis.

Comment (LinusU) [inline test/index.js]: :raised_hands:

Comment (iamstarkov) [inline test/index.js]: @LinusU fixed 😊

Comment (iamstarkov) [inline test/index.js]: @rstacruz good idea =)

**Comment by iamstarkov** [comment]
> it should, actually. did you ever figure it out?

nope, but pnpm itself is working fine with scoped modules, also on another laptop tests are fine, will investigate on problem machine with hanging tests

**Comment by iamstarkov** [comment]
unrelated to pr itself, but related to hanging tests:
1) `installCmd` doesn't play well without flags, but i [fixed][fix] it for debugging
2) everything is well with cli, as i said before
3) regarding tests, first error entry point is [`buildToStoreCached`](https://github.com/rstacruz/pnpm/blob/8f3e8f89f8c80c2e2e9183afd9136cf49ebdf650/lib/install.js#L154-L166), dunno where is the problem can be

[fix]: https://github.com/rstacruz/pnpm/commit/86dbfb916de0078b5c3f0d5b10d1edd5e2bebd0e

**Comment by iamstarkov** [comment]
anyway original pr seems to be good now

**Comment by rstacruz** [comment]
thanks!
